qiskit_nature/mappers/second_quantization/linear_mapper.py

```diff
@@ -12,16 +12,135 @@
 
 """The Linear Mapper."""
 
-from qiskit.opflow import PauliSumOp
+import operator
 
-from qiskit_nature.operators.second_quantization.spin_op import SpinOp
+from fractions import Fraction
+from functools import reduce
+from typing import List, Union
 
+import numpy as np
+
+from qiskit.opflow import PauliSumOp
+from qiskit.quantum_info.operators import Pauli, SparsePauliOp
+from qiskit_nature.operators.second_quantization.spin_op import SpinOp
 from .spin_mapper import SpinMapper
 
 
 class LinearMapper(SpinMapper):
     """The Linear spin-to-qubit mapping. """
 
     def map(self, second_q_op: SpinOp) -> PauliSumOp:
-        # TODO
-        raise NotImplementedError()
+
+        qubit_ops_list: List[PauliSumOp] = []
+
+        # get transformed general spin matrices
+        spinx, spiny, spinz, identity = self._linear_encoding(second_q_op.spin)
+
+        for idx, (_, coeff) in enumerate(second_q_op.to_list()):
+
+            operatorlist: List[PauliSumOp] = []
+
+            for n_x, n_y, n_z in zip(second_q_op.x[idx], second_q_op.y[idx], second_q_op.z[idx]):
+
+                operator_on_spin_i: List[PauliSumOp] = []
+
+                if n_x > 0:
+                    # construct the qubit operator embed
```

Comment:
Again a very minor comment. I am all for having comments but this does not seem to say much. Once can see from the code its appending to the operator_on_spin list which is defined as an operator (PauliSumOp) so it does not add any real further insight